### PR TITLE
Add --reject-cycles parameter, logging packages when found

### LINKED ISSUE #1085: Stop immediately with non-zero exit code if a cycle is found
In most cases where there's a cyclic dependency within a `lerna` project, it's a erroneous situation. Currently, we need to have our build server watch for a specific "magic string" (the warning that `lerna` emits when it discovers a cycle). Ideally, `lerna` will automatically stop its current task and exit with an error code as soon as a cycle is detected.

For existing-project-migration (and for situations where it's OK to have a cycle, if one exists?), it would be :100: if the resolution for this ticket included the ability to configure if `lerna` should fall back to its old (current) behaviour of just logging and continuing to build, even if a cycle is found.

## Expected Behavior
1. `lerna run bootstrap`
2. Lerna detects a cycle
3. Lerna stops "bootstrapping" and exits with a non-zero exit code

## Current Behavior
1. `lerna run bootstrap`
2. Lerna detects a cycle
3. Lerna warns "Encountered a cycle in the dependency graph" but keeps bootstrapping anyways.

## Possible Solution
In the same location where cycles are detected that produce the warning added by #302, instead stop the task and exit

## Steps to Reproduce (for bugs)
1. Create a `lerna` project
2. Produce a cycle within two or more packages
3. `lerna run bootstrap`

## Context
Our CI servers have to manually watch for the current warning text output when `lerna` discovers a cycle, then kills the process and stops the build. If `lerna` exited immediately with a non-zero code, then CI would work without any magic config.

## Your Environment
| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.0.0 |
| `npm --version`  | 3.10.10 |
| `node --version` | 6.11.3 |

| OS | Version |
| --- | --- |
| macOS Sierra | 10.12.6 |

### PR DESCRIPTION
**Pull request by mitchhentges**
* When `--reject-cycles` is provided, `lerna` will stop with a non-zero exit code if a cycle is detected
* When a cycle is detected, the error message lists the package names in the cycle

## Motivation and Context
See #1085

## How Has This Been Tested?
1. Created a local `lerna` project with a cycle
2. Reproduced the "error in cycle" warning without `--reject-cycles`
3. Ensured that `lerna` stops immediately if `--reject-cycles` and a cycle is found

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -893,6 +893,14 @@ May also be configured in `lerna.json`:
 }
 ```
 
+#### --reject-cycles
+
+Fail immediately if a cycle is found (in `oobstrap`, `exec`, `publish` or `run`).
+
```
Comment (bskyb-noakesj) [inline README.md]: FYI theres a typo in the word "bootstrap" both in the line above and two lines bellow.

Comment (amfio) [inline README.md]: FYI there is a typo in the word 'bootstrap' in the line above and two lines below.

Comment (mitchhentges) [reply README.md]: I swear every time I write `bootstrap` I write it differently 😆

**Comment by evocateur** [comment]
ugggghhhh appveyor ;_;

**Comment by mitchhentges** [comment]
Thanks for the approval! That was speedy quick!

**Pull request merged**